### PR TITLE
Trans Range Sensor - broken draft

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**
I can't seem to get rid of this:
```
./console/binary/tunerstudio.cpp: In function 'void startTunerStudioConnectivity()':
./console/binary/tunerstudio.cpp:864:44: error: static assertion failed: TS datapage size mismatch
  864 |  static_assert(sizeof(persistent_config_s) == TOTAL_CONFIG_SIZE, "TS datapage size mismatch");
```

### CODE REVIEW

File: firmware/integration/rusefi_config.txt [hunk 1]
```diff
@@ -1052,7 +1057,42 @@ custom maf_sensor_type_e 4 bits, S32, @OFFSET@, [0:1], @@maf_sensor_type_e_enum@
 
 	output_pin_e[LUA_PWM_COUNT iterate] luaOutputPins
 
-	int[57] unusedAtOldBoardConfigurationEnd;;"units", 1, 0, -20, 100, 0
+	#define tcu_role_e_enum "Park", "Neutral" , "Park/Neutral", "Reverse", "Drive", "Manual", "Low", "Upshift", "Downshift", "3", "2", "1"
+	custom tcu_role_e 1 bits, U08, @OFFSET@, [0:3], @@tcu_role_e_enum@@
+	tcu_role_e	tcuState0Role;
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:1062]: should this be an array?

File: firmware/integration/rusefi_config.txt [hunk 2]
```diff
@@ -1209,9 +1249,6 @@ int16_t tps2Max;Full throttle#2. tpsMax value as 10 bit ADC value. Not Voltage!\
 	brain_pin_e binarySerialRxPin;	
 	
 	brain_pin_e[AUX_DIGITAL_VALVE_COUNT iterate] auxValves
-	switch_input_pin_e tcuUpshiftButtonPin
```
Comment (mck1117) [inline firmware/integration/rusefi_config.txt:1212]: you can't remove stuff from the middle of the config, it'll change the address of everything after it (which breaks compatibility)

Comment (chuckwagoncomputing) [reply firmware/integration/rusefi_config.txt:1212]: but it didn't change the address of everything after, because of the automatic 4-byte alignment

Comment (chuckwagoncomputing) [reply firmware/integration/rusefi_config.txt:1212]: I was brute force programming at that point, and removing everything was what worked.

File: firmware/integration/rusefi_config.txt (hunk 1 above)
Comment (chuckwagoncomputing) [reply firmware/integration/rusefi_config.txt:1062]: Is this not an array?

**Comment by rusefillc** [comment]
I've cherry picked some portion of the change into master, please rebase

**Comment by rusefillc** [comment]
please rebase since some of the changes were cherry picked

File: firmware/integration/rusefi_config.txt (hunk 1 above)
Comment (mck1117) [reply firmware/integration/rusefi_config.txt:1062]: an array would look like `tcu_role_e [8 iterate] tcuStateRole;`

Comment (chuckwagoncomputing) [reply firmware/integration/rusefi_config.txt:1062]: oh right. No, it's not supposed to be an array.

Comment (mck1117) [reply firmware/integration/rusefi_config.txt:1062]: Why not? BTW adding the word "iterate" makes it spit out non-array for TS, but an array for the c++ part.

**Comment by chuckwagoncomputing** [comment]
I'm deleting this so new changes on my branch don't run actions here.

File: firmware/integration/rusefi_config.txt (hunk 1 above)
Comment (chuckwagoncomputing) [reply firmware/integration/rusefi_config.txt:1062]: oh I am an idiot

**Pull request closed without merging**